### PR TITLE
add names in some pushes

### LINKED ISSUE #394: personalize / add more info to push-notifications (e.g. "Username" invited you to connect / Your slot invitation for call "@ 1p" has been accepted)
For most of the pushes right now to understand the context you have to open the push. Let's make it more convenient for users.

@orlovvlad95 please study this library - https://hexdocs.pm/gettext/Gettext.html. The key command is `mix gettext.extract --merge`.

### PR DESCRIPTION
**Pull request by orlovvlad95**
closes #394

### CODE REVIEW

**Comment by orlovvlad95** [comment]
closes #394

File: lib/t/push_notifications/apns.ex [hunk 1]
```diff
@@ -77,10 +77,13 @@ defmodule T.PushNotifications.APNS do
...
-  def build_alert_payload("match_about_to_expire" = type, _data) do
+  def build_alert_payload("match_about_to_expire" = type, data) do
+    %{"name" => name} = data
+
     alert = %{
-      "title" => dgettext("apns", "Your match is about to expire 😢"),
-      "body" => dgettext("apns", "Invite your match to a date if you want to keep it alive ✨")
+      "title" => dgettext("apns", "Your match with %{name} is about to expire 😢", name: name),
+      "body" =>
+        dgettext("apns", "Invite your match %{name} to a date if you want to keep it alive ✨")
```
Comment (akhrail1996) [inline lib/t/push_notifications/apns.ex:86]: Putting name twice in the push doesn't sound right:

Your match with Dasha is about to expire 😢
Invite your match Dasha to a date if you want to keep it alive ✨

File: lib/t/push_notifications/apns.ex [hunk 2]
```diff
@@ -97,8 +100,10 @@ defmodule T.PushNotifications.APNS do
   end
 
   def build_alert_payload("timeslot_offer" = type, data) do
+    %{"name" => name} = data
+
     alert = %{
-      "title" => dgettext("apns", "Тебя пригласили на свидание!"),
+      "title" => dgettext("apns", "%{name} пригласил(а) тебя на свидание!", name: name),
```
Comment (akhrail1996) [inline lib/t/push_notifications/apns.ex:106]: gotta insert (a) based on gender

Comment (orlovvlad95) [reply lib/t/push_notifications/apns.ex]: done

**Comment by orlovvlad95** [comment]
@akhrail1996 
as I understand, I have to receive name or smth else before from data like that
https://github.com/getsince/test3/blob/master/lib/t/push_notifications/dispatch_job.ex#L140 
yeah?
for even DispatchJob, for example https://github.com/getsince/test3/blob/master/lib/t/matches.ex#L87

**Comment by akhrail1996** [comment]
@orlovvlad95 yes, you have to add gender to this data (where applicable)

File: priv/gettext/en/LC_MESSAGES/apns.po [hunk 3]
```diff
@@ -62,8 +62,8 @@ msgid "Твой дэйт отменён"
 msgstr "Your date has been cancelled"
 
 #, elixir-format
-#: lib/t/push_notifications/apns.ex:103
-msgid "%{name} пригласил(а) тебя на свидание!"
+#: lib/t/push_notifications/apns.ex:114
+msgid "%{name} пригласил%{gender_a} тебя на свидание!"
 msgstr "You've been asked out on a date by %{name}!"
```
Comment (akhrail1996) [inline priv/gettext/en/LC_MESSAGES/apns.po:67]: Did you test that this works as expected for both genders and for both RU & EN?

Comment (orlovvlad95) [reply priv/gettext/en/LC_MESSAGES/apns.po:67]: nope, it'll not work this way at all

**Comment by orlovvlad95** [comment]
> @orlovvlad95 yes, you have to add gender to this data (where applicable)

@akhrail1996 the problem is that (in my opinion) adding new data like name or gender maybe not applicable (without new queries, etc). I can add this, if we thought, that's important to our pushes

**Comment by akhrail1996** [comment]
@orlovvlad95 there is probably a misunderstanding - I meant to add `gender` data to the push where it will be required ("%{name} пригласил(а) тебя на свидание!").

Yeah, additional queries for gender of the notified user will probably be required for that.

File: lib/t/push_notifications/apns.ex [hunk 4]
```diff
@@ -121,8 +121,10 @@ defmodule T.PushNotifications.APNS do
   end
 
   def build_alert_payload("timeslot_accepted" = type, data) do
+    %{"name" => name} = data
+
     alert = %{
-      "title" => dgettext("apns", "Твое приглашение на дэйт принято"),
+      "title" => dgettext("apns", "Твое приглашение на дэйт c %{name} принято", name: name),
```
Comment (akhrail1996) [inline lib/t/push_notifications/apns.ex:127]: doesn't seem to work:
image

File: priv/gettext/en/LC_MESSAGES/apns.po [hunk 5]
```diff
@@ -52,8 +52,8 @@ msgid "Скорее заходи!"
 msgstr "Come look!"
 
 #, elixir-format
-#: lib/t/push_notifications/apns.ex:110 lib/t/push_notifications/apns.ex:119
-msgid "Твое приглашение на дэйт принято"
+#: lib/t/push_notifications/apns.ex:110 lib/t/push_notifications/apns.ex:126
+msgid "Твое приглашение на дэйт c %{name} принято"
 msgstr "Your invitation has been accepted"
```
Comment (akhrail1996) [inline priv/gettext/en/LC_MESSAGES/apns.po:57]: because you haven't changed it here

File: lib/t/push_notifications/dispatch_job.ex [hunk 6]
```diff
@@ -158,7 +174,6 @@ defmodule T.PushNotifications.DispatchJob do
   defp profile_name(user_id) do
     Accounts.Profile
     |> where(user_id: ^user_id)
-    |> select([p], p.name)
```
Comment (akhrail1996) [inline lib/t/push_notifications/dispatch_job.ex:161]: `profile_name` function name no longer reflects the meaning of the function

File: priv/gettext/en/LC_MESSAGES/apns.po (hunk 5 above)
Comment (orlovvlad95) [reply priv/gettext/en/LC_MESSAGES/apns.po]: fixed it

File: lib/t/push_notifications/dispatch_job.ex (hunk 6 above)
Comment (orlovvlad95) [reply lib/t/push_notifications/dispatch_job.ex:161]: changed name of function

File: lib/t/push_notifications/apns.ex (hunk 4 above)
Comment (orlovvlad95) [reply lib/t/push_notifications/apns.ex:127]: image

Comment (orlovvlad95) [reply lib/t/push_notifications/apns.ex:127]: it works now, thanks :)

File: priv/gettext/en/LC_MESSAGES/apns.po (hunk 3 above)
Comment (akhrail1996) [reply priv/gettext/en/LC_MESSAGES/apns.po:67]: does it work now?

File: lib/t/push_notifications/dispatch_job.ex (hunk 6 above)
Comment (akhrail1996) [reply lib/t/push_notifications/dispatch_job.ex:161]: I guess you only take `name` and `gender` from Profile, so probably can select just them here

File: priv/gettext/en/LC_MESSAGES/apns.po (hunk 3 above)
Comment (akhrail1996) [reply priv/gettext/en/LC_MESSAGES/apns.po:67]: image
seems to work

File: priv/gettext/ru/LC_MESSAGES/apns.po [hunk 7]
```diff
@@ -52,18 +52,18 @@ msgid "Скорее заходи!"
...
-msgid "Твой дэйт отменён"
+#: lib/t/push_notifications/apns.ex:149
+msgid "Твой дэйт с %{name} отменён"
 msgstr ""
 
 #, elixir-format
-#: lib/t/push_notifications/apns.ex:101
-msgid "Тебя пригласили на свидание!"
+#: lib/t/push_notifications/apns.ex:114
+msgid "%{name} пригласил%{gender_a} тебя на свидание!"
```
Comment (akhrail1996) [inline priv/gettext/ru/LC_MESSAGES/apns.po:66]: probably better to change it to "дэйт" everywhere

File: priv/gettext/ru/LC_MESSAGES/apns.po [hunk 8]
```diff
@@ -83,13 +83,13 @@ msgstr ""
 
 #, elixir-format
 #: lib/t/push_notifications/apns.ex:83
-msgid "Invite your match to a date if you want to keep it alive ✨"
+msgid "Invite your match %{name} to a date if you want to keep it alive ✨"
 msgstr "Пригласи на дэйт мэтча, если хочешь его сохранить ✨"
```
Comment (akhrail1996) [inline priv/gettext/ru/LC_MESSAGES/apns.po:87]: why EN has %{name} and RU doesn't?

Comment (orlovvlad95) [reply priv/gettext/ru/LC_MESSAGES/apns.po]: fixed

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 7 above)
Comment (orlovvlad95) [reply priv/gettext/ru/LC_MESSAGES/apns.po]: done

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 8 above)
Comment (akhrail1996) [reply priv/gettext/ru/LC_MESSAGES/apns.po]: image

actually both of them shouldn't have %{name}

File: priv/gettext/ru/LC_MESSAGES/apns.po [hunk 9]
```diff
@@ -63,7 +63,7 @@ msgstr ""
 
 #, elixir-format
 #: lib/t/push_notifications/apns.ex:114
-msgid "%{name} пригласил%{gender_a} тебя на свидание!"
+msgid "%{name} пригласил%{gender_a} тебя на дэйт!"
```
Comment (akhrail1996) [inline priv/gettext/ru/LC_MESSAGES/apns.po:66]: image
you cannot just change it here, you gotta change it everywhere

File: lib/t/push_notifications/dispatch_job.ex [hunk 10]
```diff
@@ -190,6 +190,7 @@ defmodule T.PushNotifications.DispatchJob do
   defp profile_info(user_id) do
     Accounts.Profile
     |> where(user_id: ^user_id)
+    |> select([p], {p.name, p.gender})
```
Comment (akhrail1996) [inline lib/t/push_notifications/dispatch_job.ex:193]: but with this the the extraction should be different, this change alone wouldn't work

previously you fetched the full %Profile{} struct and extracted gender and name like this:
image

but now you are not selecting the whole Profile struct, you are extracting just two variables in a tuple {name, gender}, therefore the extraction should be similar, something like:
{name, gender} = profile_info(user_id)
and
{name, _gender} = profile_info(user_id) - in cases where you only need names, don't forget to change them as well

Comment (akhrail1996) [reply lib/t/push_notifications/dispatch_job.ex:193]: image

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 8 above)
Comment (orlovvlad95) [reply priv/gettext/ru/LC_MESSAGES/apns.po]: agree with you, deleted one of these

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 9 above)
Comment (orlovvlad95) [reply priv/gettext/ru/LC_MESSAGES/apns.po:66]: done

File: lib/t/push_notifications/dispatch_job.ex (hunk 10 above)
Comment (orlovvlad95) [reply lib/t/push_notifications/dispatch_job.ex:177]: thanks for good review, changed using of struct like you said

**Comment by orlovvlad95** [comment]
@akhrail1996 could you pls help me with failing test. It doesn't reproduce in local test, I haven't any failures. Do you have any idea what's going wrong?

File: lib/t/push_notifications/dispatch_job.ex [hunk 11]
```diff
@@ -33,17 +33,13 @@ defmodule T.PushNotifications.DispatchJob do
     if match = alive_match(match_id) do
       %Matches.Match{user_id_1: uid1, user_id_2: uid2} = match
 
-      if profile1 = profile_info(uid1) do
-        %Accounts.Profile{name: name1} = profile1
-        data1 = %{"match_id" => match_id, "name" => name1}
-        uid1 |> Accounts.list_apns_devices() |> schedule_apns("match_about_to_expire", data1)
-      end
+      {name1, _gender} = profile_info(uid1)
```
Comment (akhrail1996) [inline lib/t/push_notifications/dispatch_job.ex:36]: "if" didn't need to be deleted:
if profile1 = profile_info(uid1) do
      {name1, _gender} = profile1

Comment (akhrail1996) [reply lib/t/push_notifications/dispatch_job.ex:36]: (same in all other cases below)

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 8 above)
Comment (akhrail1996) [reply priv/gettext/ru/LC_MESSAGES/apns.po]: image
it's still there

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 9 above)
Comment (akhrail1996) [reply priv/gettext/ru/LC_MESSAGES/apns.po:66]: image
there are still "свидание" left
and please do not forget to run `mix gettext.extract --merge` every time when you edit some of gettext texts

**Comment by akhrail1996** [comment]
> @akhrail1996 could you pls help me with failing test. It doesn't reproduce in local test, I haven't any failures. Do you have any idea what's going wrong?

yeah, if you look at the failing test you can see that it's "test more with distance filter":
image

if you try to locate this test in your branch (prior to my last merge commit), you wouldn't find it -> this test was added to master branch after you "branched out" from master with your `add_info_in_pushes` branch. Therefore it's pretty logical that this test fails (as the part of code being tested in this test does not exist in your branch).

It's not very intuitive that GitHub Actions run tests from master branch on your PR, but after I thought about it - this starts to make sense as your PR is into master, and therefore to test whether it's good to be merged into master, GitHub Actions is testing it on master tests. 

I guess a good practice for this going forward would be to merge new work from master locally in PR when such problems occur (the same way i did  here).

**Comment by orlovvlad95** [comment]
> > @akhrail1996 could you pls help me with failing test. It doesn't reproduce in local test, I haven't any failures. Do you have any idea what's going wrong?
> 
> yeah, if you look at the failing test you can see that it's "test more with distance filter": image
> 
> if you try to locate this test in your branch (prior to my last merge commit), you wouldn't find it -> this test was added to master branch after you "branched out" from master with your `add_info_in_pushes` branch. Therefore it's pretty logical that this test fails (as the part of code being tested in this test does not exist in your branch).
> 
> It's not very intuitive that GitHub Actions run tests from master branch on your PR, but after I thought about it - this starts to make sense as your PR is into master, and therefore to test whether it's good to be merged into master, GitHub Actions is testing it on master tests.
> 
> I guess a good practice for this going forward would be to merge new work from master locally in PR when such problems occur (the same way i did here).

@akhrail1996  ok, thanks for info and merge

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 9 above)
Comment (orlovvlad95) [reply priv/gettext/ru/LC_MESSAGES/apns.po:66]: ok, sorry, my mistake

File: lib/t/push_notifications/dispatch_job.ex (hunk 11 above)
Comment (orlovvlad95) [reply lib/t/push_notifications/dispatch_job.ex]: done

File: priv/gettext/ru/LC_MESSAGES/apns.po (hunk 8 above)
Comment (orlovvlad95) [reply priv/gettext/ru/LC_MESSAGES/apns.po]: seems like I forget to commit, sorry again

**Comment by orlovvlad95** [comment]
@akhrail1996  do not understand why these conflicts happened
run mix gettext.extract --merge as you said

**Comment by akhrail1996** [comment]
> @akhrail1996 do not understand why these conflicts happened run mix gettext.extract --merge as you said

This is expected since I merged another PR, in which there was a new push-notification added, and an old one deleted: https://github.com/getsince/test3/commit/2dd40ade3ddff2890483919a2764934e5b0691c1.
So some lines naturally changed (deletion of old notification https://github.com/getsince/test3/commit/2dd40ade3ddff2890483919a2764934e5b0691c1#diff-817affbf31ea787ee999027181555c08968180038042b914503f6bd60aa970e8L95 caused other lines in the file "go up" -> apns.pot / po files changed).

So please resolve the conflicts accordingly and run `mix gettext.extract --merge` again, and check that lines are correct.

**Review by akhrail1996 — APPROVED** [review]
all good 👍
pls squash & merge and deploy

**Pull request merged**